### DISCUSSION TITLE
[Contract Verification] How to verify upgradeable contract(TransparentUpgradeableProxy) on zkSync by Hardhat?

### DISCUSSION THREAD

**Discussion by RomanYarinski** [zkSync-Community-Hub/zksync-developers#417] [Contract Verification] [answered]
### Environment

Mainnet

### Issue Type

- [X] Plugin
- [ ] Block Explorer

### Contract Address

0xf7483A1464DeF6b8d5A6Caca4A8ce7E5be8F1F68

### Compiler Type

Multi-part contract

### zkSolc Version

0.3.18

### Solc Version

8.0.19

### Contract Name

TransparentUpgradeableProxy

### Contract Code

Standard TransparentUpgradeableProxy

### Constructor Arguments

Standard TransparentUpgradeableProxy params, initialize method without arguments

### Additional Details

```
hh verify --network zkSyncMainnet 0xf7483A1464DeF6b8d5A6Caca4A8ce7E5be8F1F68
EVM ENV
Verifying implementation: 0x33A6A91f601Ffc837AA831a576E8798C9F26096F
The zksolc compiler version in your Hardhat config file (1.3.18) is not the latest. We recommend using the latest version 1.4.0.
Compiling contracts for zkSync Era with zksolc v1.3.18 and solc v0.8.19
Compiling 22 Solidity files
Implementation on address 0x33A6A91f601Ffc837AA831a576E8798C9F26096F already verified.
Verifying proxy: 0xf7483A1464DeF6b8d5A6Caca4A8ce7E5be8F1F68
No logs found for event topic AdminChanged(address,address) at address 0xf7483A1464DeF6b8d5A6Caca4A8ce7E5be8F1F68
One of possible reasons can be that you are trying to verify a UUPS contract
No logs found for event topic Upgraded(address) at address 0xf7483A1464DeF6b8d5A6Caca4A8ce7E5be8F1F68
One of possible reasons can be that you are trying to verify a UUPS contract
Could not find an event with any of the following topics in the logs for address 0xf7483A1464DeF6b8d5A6Caca4A8ce7E5be8F1F68: AdminChanged(address,address), Upgraded(address)If the proxy was recently deployed, the transaction may not be available on Block Explorer yet. Try running the verify task again after waiting a few blocks.
Verifying proxy admin: 0xDf4a5F0FC2B5524dF89e0BC134f955a9fF348422
No logs found for event topic OwnershipTransferred(address,address) at address 0xDf4a5F0FC2B5524dF89e0BC134f955a9fF348422
One of possible reasons can be that you are trying to verify a UUPS contract
Could not find an event with any of the following topics in the logs for address 0xDf4a5F0FC2B5524dF89e0BC134f955a9fF348422: OwnershipTransferred(address,address)If the proxy was recently deployed, the transaction may not be available on Block Explorer yet. Try running the verify task again after waiting a few blocks.
```

Please advise how I can verify the proxy contract? Admin and Implementation contracts are verified without problems.

**Comment by bxpana** [comment]
Did you already try the steps listed in our docs for verifying proxies? https://docs.zksync.io/build/tooling/hardhat/hardhat-zksync-upgradable.html#upgrade-beacon-proxy

**Reply by RomanYarinski** [reply]
Yes, of course. Tried it with different versions of the library. 
I also tried verifying on direct using the example from Openzeppeling https://forum.openzeppelin.com/t/how-to-verify-upgradeable-contract-on-opbnb-testnet-by-hardhat/39495/2 for another network.
Can you advise me on possible variants?

**Reply by mpopovac-txfusion** [reply]
Hey @RomanYarinski , could you provide more details about your setup, such as the version of the @openzeppelin/contracts package you're using, etc? Even better, if you could share a link to the repository, it would be great so we can take a closer look.

**Reply by maxfundingrate** [reply]
@mpopovac-txfusion just invited you to our repo (Vest Labs)

**Reply by mpopovac-txfusion** [reply]
Hey @maxfundingrate , thanks for the invite. I inspected the problem and it's not connected directly with the plugin itself, rather to SDK (newer versions for v5) used for sending requests to obtain logs, which are required to find the appropriate contract for verification. It appears that the SDK doesn't handle "falsy" values correctly when creating params for `eth_getLogs` RPC method.

I will contribute to the SDK to address this issue, but I will also double-check on our plugin side to ensure we avoid this problem. I will notify you when a new release for the upgradable plugin is available.

Additionally, I've verified your contract :D : https://explorer.zksync.io/address/0xf7483A1464DeF6b8d5A6Caca4A8ce7E5be8F1F68#contract

**Reply by RomanYarinski** [reply]
@mpopovac-txfusion 
This is fantastic news, thank you :D
Can we ask you to verify our second proxy contract as well? Exactly the same Proxy for another implementation.
https://explorer.zksync.io/address/0xa8412d1cDbd9cde358cF382FF0D6e88ecA7d7d20#transactions

**Reply by mpopovac-txfusion** [reply, marked as answer]
It is done. :D
